pallets/ah-migrator/src/lib.rs

```diff
@@ -590,32 +550,32 @@ pub mod pallet {
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			let res = Self::do_receive_fast_unstake_messages(messages);
-
-			Self::increment_msg_received_count(res.is_err());
-
-			res.map_err(Into::into)
+			Self::do_receive_fast_unstake_messages(messages).map_err(Into::into)
 		}
 
 		/// Receive referendum counts, deciding counts, votes for the track queue.
 		#[pallet::call_index(10)]
 		#[pallet::weight(T::AhWeightInfo::receive_referenda_values())]
 		pub fn receive_referenda_values(
 			origin: OriginFor<T>,
-			referendum_count: u32,
-			// track_id, count
-			deciding_count: Vec<(TrackIdOf<T, ()>, u32)>,
-			// referendum_id, votes
-			track_queue: Vec<(TrackIdOf<T, ()>, Vec<(u32, u128)>)>,
+			mut values: Vec<(
+				// referendum_count
+				u32,
+				// deciding_count (track_id, count)
+				Vec<(TrackIdOf<T, ()>, u32)>,
+				// track_queue (referendum_id, votes)
+				Vec<(TrackIdOf<T, ()>, Vec<(u32, u128)>)>,
+			)>,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			let res =
-				Self::do_receive_referenda_values(referendum_count, deciding_count, track_queue);
+			ensure!(values.len() == 1, Error::<T>::InvalidParameters);
```

Comment:
Why do you use a vec then?

Reply:
bc of the `send_chunked_xcm_and_track` signature. I used `send_xcm` before for it. but since only the `send_chunked_xcm_and_track` has tracking I have to use it and I did not want to change the working code a lot. the `receive_referenda_values` we designed first only to receive a single message, not a batch.